module/zfs/metaslab.c

```diff
@@ -850,9 +850,17 @@ metaslab_group_create(metaslab_class_t *mc, vdev_t *vd, int allocators)
 		metaslab_group_allocator_t *mga = &mg->mg_allocator[i];
 		zfs_refcount_create_tracked(&mga->mga_alloc_queue_depth);
 	}
-
+#ifdef __APPLE__
+#ifndef _KERNEL
+#define	TASKQ_REALLY_DYNAMIC 0x0
+#endif
+	mg->mg_taskq = taskq_create("metaslab_group_taskq", metaslab_load_pct,
+	    maxclsyspri, 10, INT_MAX, TASKQ_THREADS_CPU_PCT | TASKQ_DYNAMIC
+	    | TASKQ_REALLY_DYNAMIC);
```

Comment:
That is a bit unattractive, should see if we can move it elsewhere. In this case, it's for the same code compiled in userland mode, where that bit doesn't make sense/isn't needed.

Reply:
There's a bit of history here too, Solaris/illumos taskq considers it invalid to use `TASKQ_DYNAMIC` with `dispatch_ent`. (And checks for it).
Linux decided it was OK, or perhaps to mean something slightly different, and use it liberally.
So we made `TASKQ_DYNAMIC` to be nothing, so it will still work with our Solaris taskq.
Then we realised we really do want a version of `TASKQ_DYNAMIC` in the macOS, in a couple of places.

I've changed that conditional to the more sane: 

```
#if defined(__APPLE__) && defined(_KERNEL)
```